### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by Liubomyr19**
[DEMO LINK](https://Liubomyr19.github.io/layout_calendar/)
[TEST REPORT LINK](https://Liubomyr19.github.io/layout_calendar/report/html_report/)

### CODE REVIEW

File: src/styles/main.scss [hunk 1]
```diff
@@ -1,3 +1,60 @@
...
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  margin: auto;
+  width: 721px;
+  padding: 10px;
+  gap: 1px;
+
+  &__field {
+    background-color: #eee;
```
Comment (loralevitska) [inline src/styles/main.scss]: It is a good practice to make all colors variable

File: src/styles/main.scss [hunk 2]
```diff
@@ -1,3 +1,60 @@
...
+    height: 100px;
+    width: 100px;
+    border: solid black 1px;
+    font-size: 30px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    transform: all 0.5s;
```
Comment (loralevitska) [inline src/styles/main.scss]:
```suggestion
    transition: transform 0.5s;
```

File: src/styles/main.scss [hunk 3]
```diff
@@ -1,3 +1,60 @@
...
+  }
+
+  @for $month-length from 28 through 31 {
+    &--month-length--#{$month-length}
+    .calendar__field:nth-child(n + #{$month-length + 1}) {
+      display: none;
+    }
+  }
+
+  $start-day: "mon" 0,"tue" 1,"wed" 2,"thu" 3,"fri" 4,"sat" 5,"sun" 6;
```
Comment (loralevitska) [inline src/styles/main.scss]: Move variables to separate file

**Review by loralevitska — CHANGES_REQUESTED** [review]
Good job! There are things that should be fixed:

- fix month-length modifier 
image

File: src/styles/main.scss [hunk 4]
```diff
@@ -1,3 +1,63 @@
+@import url(_variables.scss);
+
+$field--background-color: #eee;
+$field--hover-color: pink;
```
Comment (DarkMistyRoom) [inline src/styles/main.scss:4]: move them to the `variables` file

File: src/styles/main.scss [hunk 5]
```diff
@@ -1,3 +1,63 @@
...
   margin: 0;
 }
+
+.main {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+  font-family: Arial, Helvetica, sans-serif;
+  margin: 9px;
```
Comment (DarkMistyRoom) [inline src/styles/main.scss:16]: Don't add external styles (positioning or margins) to BEM-blocks. Use mix where necessary and move all external styles under element selector.

**Review by DarkMistyRoom — CHANGES_REQUESTED** [review]
- A calendar should always have 7 columns

image

- fix the modifiers. length doesn't work. start day applies wrong indent
image

- also check the syntax. it should be like `calendar--start-day--28`

- no need to compile these files to src
image

File: src/styles/main.scss (hunk 4 above)
Comment (VitaliyBodnar) [reply src/styles/main.scss:4]: not fixed

File: src/styles/main.scss (hunk 5 above)
Comment (VitaliyBodnar) [reply src/styles/main.scss:16]: not fixed

**Review by VitaliyBodnar — CHANGES_REQUESTED** [review]
- month-length modifier still not working
Screenshot 2023-07-15 at 15 10 51

**Review by SanyaBratashchuk — CHANGES_REQUESTED** [review]
Remove extra cell 
image

File: src/styles/main.scss (hunk 4 above)
Comment (loralevitska) [inline src/styles/main.scss:4]: According previous comment you should move this variables to your file `_variables.scss` 
image

File: src/styles/main.scss (hunk 5 above)
Comment (loralevitska) [inline src/styles/main.scss:16]: Improve it 
image
```suggestion
```

Use margins for mix class or for body tag in your situation

**Review by loralevitska — CHANGES_REQUESTED** [review]
Hi, I see you haven't fixed previous comments, maybe you did not push your changes? If you have any questions, please ask them in fe_chat and tag mentors. It will be faster, because your answers in GitHub will be visible to mentor only after you re-request review

**Pull request closed without merging**